ballista/rust/scheduler/src/state/mod.rs

```diff
@@ -82,237 +85,639 @@ pub enum WatchEvent {
     Delete(String),
 }
 
+type JobTasks = HashMap<u32, HashMap<u32, TaskStatus>>;
+
 #[derive(Clone)]
-pub(super) struct SchedulerState {
+struct VolatileSchedulerState {
+    executors_heartbeat: Arc<std::sync::RwLock<HashMap<String, ExecutorHeartbeat>>>,
```

Comment:
Should we use RwLock from parking_lot ?

Reply:
Agree. Should be consistent with other parts. And the performance may also be better.